dapps/frenemies/src/utils/coins.ts

```diff
@@ -1,76 +1,58 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { Coin, SUI_COIN } from "../network/types";
-import { ObjectData } from "../network/rawObject";
-import { getTransactionEffects } from "@mysten/sui.js";
+import {
+  CoinStruct,
```

Comment:
I guess we could extend this buddy and turn into `CoinStructSafe` with bigint instead.